### PR TITLE
Refactor: backward-compatible events

### PR DESCRIPTION
**Pull request by katspaugh**
Refactor the event system to be backward compatible with v6 event signatures.
Also added `un` and `unAll` like in v6.

### CODE REVIEW

File: src/event-emitter.ts [hunk 1]
```diff
@@ -1,39 +1,62 @@
 export interface GeneralEventTypes {
   // the name of the event and the data it dispatches with
-  // e.g. 'entryCreated': { count: 1 }
-  [eventType: string]: unknown
+  // e.g. 'entryCreated': [count: 1]
+  [eventType: string]: Array<any> // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (katspaugh) [inline src/event-emitter.ts:4]: I haven't figured out yet how to type this properly.
If we don't type the event data as any here, the WaveSurfer class complains that actual plugins are incompatible with GeneralPlugin.

**Comment by katspaugh** [comment]
Gonna merge this, but still looking for feedback

**Comment by prakharpbuf** [comment]
Hi Katspaugh,

Thanks for reaching out and asking for my feedback, I really appreciate your willingness to involve me in the project and I'm excited to continue contributing to wavesurfer.ts.

However, I want to be upfront with you and let you know that I'm not yet familiar with TypeScript or object-oriented JavaScript, so the code you've written is a bit too complex for me to fully understand and provide a comprehensive review.

That being said, I'm eager to learn and improve my skills, and I'm definitely interested in continuing to contribute to the project. If you have any resources or materials you could recommend for learning TypeScript and object-oriented JavaScript, I'd really appreciate it.

Thanks again.

**Comment by prakharpbuf** [comment]
One more thing, I haven't yet started using the alpha version of wavesurfer because the DAW project I was working on, is on hold momentarily, and I'm working on voice cloning deep learning tech. Checkout [coqui.ai](https://github.com/coqui-ai/TTS) [Real Time Voice Cloning](https://github.com/CorentinJ/Real-Time-Voice-Cloning)

I expect to continue that project soon and maybe then I will have better feedback.

**Comment by katspaugh** [comment]
Sure, I understand. Coqui.ai and voice cloning look cool!
@gartenfeld you might be interested too.

I'm not even a fan of OOP myself, and am also still learning the more advanced TypeScript features, mostly on a case-by-case basis and with the help of ChatGPT. :) And by following this gentleman on Twitter: https://twitter.com/mattpocockuk

I appreciate all your help on this project!

**Pull request merged**